beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -443,3 +446,138 @@ func (s *Server) ListBLSToExecutionChanges(w http.ResponseWriter, r *http.Reques
 		Data: changes,
 	})
 }
+
+// GetAttesterSlashings retrieves attester slashings known by the node but
+// not necessarily incorporated into any block.
+func (s *Server) GetAttesterSlashings(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetAttesterSlashings")
+	defer span.End()
+
+	headState, err := s.ChainInfoFetcher.HeadStateReadOnly(ctx)
+	if err != nil {
+		http2.HandleError(w, "Could not get head state: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	sourceSlashings := s.SlashingsPool.PendingAttesterSlashings(ctx, headState, true /* return unlimited slashings */)
+	slashings := shared.AttesterSlashingsFromConsensus(sourceSlashings)
+
+	http2.WriteJson(w, &GetAttesterSlashingsResponse{Data: slashings})
+}
+
+// SubmitAttesterSlashing submits AttesterSlashing object to node's pool and
+// if passes validation node MUST broadcast it to network.
+func (s *Server) SubmitAttesterSlashing(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.SubmitAttesterSlashing")
+	defer span.End()
+
+	var req shared.AttesterSlashing
+	err := json.NewDecoder(r.Body).Decode(&req)
+	switch {
+	case err == io.EOF:
```

Comment:
i think you only need io.EOF if it's an array perhaps i messed up some older endpoints if that's not the case i need to double check.

Reply:
Even if it's the case, it's very hard to know that when writing a new endpoint. I would just keep this case everywhere as it doesn't hurt.